### PR TITLE
feat: all dashboard is migrated to useQuery and action is removed

### PR DESCRIPTION
**Pull request by himanshugupta714**
video: https://www.loom.com/share/70a8628a2c244a9f8fbe117e45de1fd1

part of https://github.com/SigNoz/signoz/issues/2107

### CODE REVIEW

**Comment by himanshugupta714** [comment]
@YounixM @palashgdev can you please check

**Comment by palashgdev** [comment]
thanks for sharing the PR @himanshugupta714 .. let me check

File: frontend/src/api/dashboard/getAll.ts [hunk 1]
```diff
@@ -1,24 +1,8 @@
...
-	} catch (error) {
-		return ErrorResponseHandler(error as AxiosError);
-	}
-};
-
-export default getAll;
+export const getAllDashboardList = (): Promise<Dashboard[]> =>
+	axios
+		.get<ApiResponse<Dashboard[]>>('/dashboards')
+		.then((res) => res.data.data);
```
Comment (Damans227) [inline frontend/src/api/dashboard/getAll.ts:8]:
```
		.then((res) => res.data.data);
```

`res` is too generic, we can use a better name - something closer to the type of value that `res` holds.

File: frontend/src/container/ListOfDashboard/TableComponents/DeleteButton.tsx [hunk 2]
```diff
@@ -20,13 +24,14 @@ function DeleteButton({ deleteDashboard, id }: DeleteButtonProps): JSX.Element {
 			onOk() {
 				deleteDashboard({
 					uuid: id,
+					refetch: refetchDashboardList,
```
Comment (Damans227) [inline frontend/src/container/ListOfDashboard/TableComponents/DeleteButton.tsx:27]:
```suggestion
					reFetch: reFetchDashboardList,
```

File: frontend/src/container/ListOfDashboard/TableComponents/DeleteButton.tsx [hunk 3]
```diff
@@ -20,13 +24,14 @@ function DeleteButton({ deleteDashboard, id }: DeleteButtonProps): JSX.Element {
...
 				});
 			},
 			okText: 'Delete',
 			okButtonProps: { danger: true },
 			centered: true,
 		});
-	}, [id, modal, deleteDashboard]);
+	}, [modal, deleteDashboard, id, refetchDashboardList]);
```
Comment (Damans227) [inline frontend/src/container/ListOfDashboard/TableComponents/DeleteButton.tsx:34]: `id` isn't a good variable name by itself.May be, `dashboardId` ?

File: frontend/src/container/ListOfDashboard/TableComponents/DeleteButton.tsx [hunk 4]
```diff
@@ -51,13 +56,22 @@ const mapDispatchToProps = (
...
 
 const WrapperDeleteButton = connect(null, mapDispatchToProps)(DeleteButton);
 
 // This is to avoid the type collision
 function Wrapper(props: Data): JSX.Element {
-	const { createdBy, description, id, key, lastUpdatedTime, name, tags } = props;
+	const {
+		createdBy,
+		description,
+		id,
```
Comment (Damans227) [inline frontend/src/container/ListOfDashboard/TableComponents/DeleteButton.tsx:68]: same as above.

File: frontend/src/container/ListOfDashboard/TableComponents/DeleteButton.tsx [hunk 5]
```diff
@@ -51,13 +56,22 @@ const mapDispatchToProps = (
...
+		key,
+		refetchDashboardList,
+		lastUpdatedTime,
+		name,
```
Comment (Damans227) [inline frontend/src/container/ListOfDashboard/TableComponents/DeleteButton.tsx:72]: What does the name represent here? Dashboard name?

File: frontend/src/api/dashboard/getAll.ts (hunk 1 above)
Comment (palashgdev) [reply frontend/src/api/dashboard/getAll.ts:8]: @Damans227 can you suggest seems @himanshugupta714 is trying to highlight `response`

File: frontend/src/container/ListOfDashboard/TableComponents/DeleteButton.tsx (hunk 3 above)
Comment (palashgdev) [reply frontend/src/container/ListOfDashboard/TableComponents/DeleteButton.tsx:34]: good suggestion @Damans227 though this is not related to this PR

File: frontend/src/container/ListOfDashboard/TableComponents/DeleteButton.tsx (hunk 5 above)
Comment (palashgdev) [reply frontend/src/container/ListOfDashboard/TableComponents/DeleteButton.tsx:72]: name of the dashboard

**Review by YounixM — APPROVED** [review]

**Pull request merged**